opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -14,124 +14,185 @@
+
+    def aggregate(self, measurement: Measurement):
+        with self._lock:
+            self._value = measurement.value
 
+    def collect(self) -> Optional[_PointVarT]:
+        """
+        Atomically return a point for the current value of the metric.
+        """
+        if self._value is None:
+            return None
 
-class ExplicitBucketHistogramAggregation(Aggregation):
+        return Gauge(
+            time_unix_nano=_time_ns(),
+            value=self._value,
+        )
 
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
 
+class ExplicitBucketHistogramAggregation(Aggregation[Histogram]):
     def __init__(
         self,
-        instrument,
-        *args,
-        boundaries=(0, 5, 10, 25, 50, 75, 100, 250, 500, 1000),
-        record_min_max=True,
+        is_monotonic: bool,
+        boundaries: Sequence[int] = (
+            0,
+            5,
+            10,
+            25,
+            50,
+            75,
+            100,
+            250,
+            500,
+            1000,
+        ),
+        record_min_max: bool = True,
     ):
-        super().__init__()
+        super().__init__(is_monotonic)
         self._value = OrderedDict([(key, 0) for key in (*boundaries, inf)])
         self._min = inf
         self._max = -inf
         self._sum = 0
-        self._instrument = instrument
         self._record_min_max = record_min_max
+        self._start_time_unix_nano = _time_ns()
+        self._boundaries = boundaries
 
-    @property
-    def min(self):
-        if not self._record_min_max:
-            _logger.warning("Min is not being recorded")
-
-        return self._min
+    def aggregate(self, measurement: Measurement) -> None:
 
-    @property
-    def max(self):
-        if not self._record_min_max:
-            _logger.warning("Max is not being recorded")
+        value = measurement.value
 
-        return self._max
-
-    @property
-    def sum(self):
-        if isinstance(self._instrument, _Monotonic):
-            return self._sum
-
-        _logger.warning(
-            "Sum is not filled out when the associated "
-            "instrument is not monotonic"
-        )
-        return None
-
-    def aggregate(self, value):
         if self._record_min_max:
             self._min = min(self._min, value)
             self._max = max(self._max, value)
 
-        if isinstance(self._instrument, _Monotonic):
+        if self._is_monotonic:
             self._sum += value
 
         for key in self._value.keys():
 
             if value < key:
-                self._value[key] = self._value[key] + value
+                self._value[key] = self._value[key] + 1
 
                 break
 
-    def make_point_and_reset(self):
-        pass
+    def collect(self) -> Optional[_PointVarT]:
+        """
+        Atomically return a point for the current value of the metric.
+        """
+        now = _time_ns()
+
+        with self._lock:
+            self._value = 0
```

Comment:
Can you also add tests for `collect` for all aggregations? Also same as above about saving before resetting.
```suggestion
            self._value = OrderedDict([(key, 0) for key in (*boundaries, inf)])
```

Reply:
Added tests for collect as well :+1: